agent/internal/containers.go

```diff
@@ -144,32 +145,35 @@ func (c *containerManager) overwriteSpec(
 		slotIds = append(slotIds, strconv.Itoa(d.ID))
 	}
 	spec.RunSpec.ContainerConfig.Labels[dockerContainerDevicesLabel] = strings.Join(slotIds, ",")
-	if len(cont.Devices) > 0 && cont.Devices[0].Type == device.GPU {
-		spec.RunSpec.HostConfig.Runtime = nvidiaRuntime
+
+	spec = c.overwriteSpecforGPUs(cont, spec)
```

Comment:
non-blocking: think we may want to do something like this instead to match what it looks like for env vars

```	
spec.RunSpec.HostConfig.DeviceRequests = append(
		spec.RunSpec.HostConfig.DeviceRequests, c.gpuDevices(cont)...)
```